pkg/trojan/message_test.go

```diff
@@ -0,0 +1,248 @@
+		t.Fatalf("expected error when creating message of invalid payload size to be %q, but got %v", ErrPayloadTooBig, err)
+	}
+}
+
+// TestWrap tests the creation of a chunk from a list of targets
+// its address length and span should be correct
+// its resulting address should have a prefix which matches one of the given targets
+// its resulting data should have a hash that matches its address exactly
+func TestWrap(t *testing.T) {
+	m := newTestMessage(t)
+	c, err := m.Wrap(testTargets)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	addr := c.Address()
+	addrLen := len(addr.Bytes())
+	if addrLen != swarm.HashSize {
+		t.Fatalf("chunk has an unexpected address length of %d rather than %d", addrLen, swarm.HashSize)
+	}
+
+	addrPrefix := addr.Bytes()[:len(testTargets[0])]
+	if !contains(testTargets, addrPrefix) {
+		t.Fatal("chunk address prefix does not match any of the targets")
+	}
+
+	data := c.Data()
+	dataSize := len(data)
+	expectedSize := 8 + swarm.ChunkSize // span + payload
+	if dataSize != expectedSize {
+		t.Fatalf("chunk data has an unexpected size of %d rather than %d", dataSize, expectedSize)
+	}
+
+	span := binary.LittleEndian.Uint64(data[:8])
+	remDataLen := len(data[8:])
+	if int(span) != remDataLen {
+		t.Fatalf("chunk span set to %d, but rest of chunk data is of size %d", span, remDataLen)
+	}
+
+	dataHash, err := hashBytes(data)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if !bytes.Equal(addr.Bytes(), dataHash) {
+		t.Fatal("chunk address does not match its data hash")
+	}
+}
+
+// TestWrapFail tests that the creation of a chunk fails when given targets are invalid
+func TestWrapFail(t *testing.T) {
+	m := newTestMessage(t)
+
+	emptyTargets := Targets([]Target{})
+	if _, err := m.Wrap(emptyTargets); err != ErrEmptyTargets {
+		t.Fatalf("expected error when creating chunk for empty targets to be %q, but got %v", ErrEmptyTargets, err)
+	}
+
+	t1 := Target([]byte{34})
+	t2 := Target([]byte{25, 120})
+	t3 := Target([]byte{180, 18, 255})
+	varLenTargets := Targets([]Target{t1, t2, t3})
+	if _, err := m.Wrap(varLenTargets); err != ErrVarLenTargets {
+		t.Fatalf("expected error when creating chunk for variable-length targets to be %q, but got %v", ErrVarLenTargets, err)
+	}
+}
+
+// TestPadBytes tests that different types of byte slices are correctly padded with leading 0s
+// all slices are interpreted as big-endian
+func TestPadBytes(t *testing.T) {
+	s := make([]byte, 32)
+
+	// empty slice should be unchanged
+	p := padBytes(s)
+	if !bytes.Equal(p, s) {
+		t.Fatalf("expected byte padding to result in %x, but is %x", s, p)
+	}
+
+	// slice of length 3
+	s = []byte{255, 128, 64}
+	p = padBytes(s)
+	e := append(make([]byte, 29), s...) // 29 zeros plus the 3 original bytes
+	if !bytes.Equal(p, e) {
+		t.Fatalf("expected byte padding to result in %x, but is %x", e, p)
+	}
+
+	// simulate toChunk behavior
+	s = []byte{255, 255, 255}
+	i := new(big.Int).SetBytes(s) // byte slice to big.Int
```

Comment:
Here it looks like you're testing whether `big.Int` interprets bytes correctly. That doesn't seem to belong here.

Reply:
indeed, this can probably be replaced by directly using `s = []byte{1, 0, 0, 0}`